internal/authz/authz.go

```diff
@@ -169,6 +191,65 @@ func (a *Authorizer) AuthorizeCreateStore(ctx context.Context) error {
 	return a.individualAuthorize(ctx, claims.ClientID, relation, System, &openfgav1.ContextualTupleKeys{})
 }
 
+// GetModulesForWriteRequest returns the modules that should be checked for the write request.
+// If we encounter a type with no attached module, we should break and return no modules so that the authz check will be against the store
+// Otherwise we return a list of unique modules encountered so that FGA on FGA can check them after.
+func (a *Authorizer) GetModulesForWriteRequest(req *openfgav1.WriteRequest, typesys *typesystem.TypeSystem) ([]string, error) {
+	modulesMap := make(map[string]struct{})
+
+	modulesMap, shouldCheckOnStore, err := processTupleKeys(req.GetWrites().GetTupleKeys(), typesys, modulesMap)
+	if err != nil {
+		return nil, err
+	}
+
+	if !shouldCheckOnStore {
+		modulesMap, shouldCheckOnStore, err = processTupleKeys(req.GetDeletes().GetTupleKeys(), typesys, modulesMap)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if shouldCheckOnStore {
+		return []string{}, nil
+	}
+
+	modules := make([]string, 0, len(modulesMap))
+	for module := range modulesMap {
+		modules = append(modules, module)
+	}
```

Comment:
non-blocking: pro-tip: if you know the length of the slice up-front, and you will populate each index of the slice, it is much more efficient to use direct index access to populate the array, instead of using `append()`. 
```go
modules := make([]string, len(modulesMap))
var i int
for module := range modulesMap {
    modules[i] = module
    i++
}
```

Reply:
Cool! Thanks